obspy/io/reftek/core.py

```diff
@@ -0,0 +1,341 @@
+    bytes) and checks for valid packet type identifiers in the first 20
+    expected packet positions.
+    """
+    if not os.path.isfile(filename):
+        return False
+    filesize = os.stat(filename).st_size
+    # check if overall file size is a multiple of 1024
+    if filesize < 1024 or filesize % 1024 != 0:
+        return False
+
+    with open(filename, 'rb') as fp:
+        # check first 20 expected packets' type header field
+        while True:
+            packet_type = fp.read(2).decode("ASCII", "ignore")
+            if not packet_type:
+                break
+            if packet_type not in PACKET_TYPES:
+                return False
+            fp.seek(1022, os.SEEK_CUR)
+    return True
+
+
+def _read_reftek130(filename, network="", location="", component_codes=None,
+                    headonly=False, verbose=False, **kwargs):
+    """
+    Read a REFTEK130 file into an ObsPy Stream.
+
+    :type filename: str
+    :param filename: REFTEK130 file to be read.
+    :type network: str
+    :param network: Network code to fill in for all data (network code is not
+        stored in EH/ET/DT packets).
+    :type location: str
+    :param location: Location code to fill in for all data (network code is not
+        stored in EH/ET/DT packets).
+    :type component_codes: list
+    :param component_codes: Iterable of single-character component codes (e.g.
+        ``['Z', 'N', 'E']``) to be appended to two-character stream name parsed
+        from event header packet (e.g. ``'HH'``) for each of the channels in
+        the data (e.g. to make the channel codes in a three channel data file
+        to ``'HHZ'``, ``'HHN'``, ``'HHE'`` in the created stream object).
+    :type headonly: bool
+    :param headonly: Determines whether or not to unpack the data or just
+        read the headers.
+    :rtype: :class:`~obspy.core.stream.Stream`
+    """
+    # Reftek 130 data format stores only the last two digits of the year.  We
+    # currently assume that 00-49 are years 2000-2049 and 50-99 are years
+    # 2050-2099. We deliberately raise an exception if the current year will
+    # become 2050 (just in case someone really still uses this code then.. ;-)
+    # At that point the year would probably have to be explicitly specified
+    # when reading data to be on the safe side.
+    if NOW.year > 2050:
+        raise NotImplementedError()
+    return Reftek130.from_file(filename).to_stream(
+        network=network, location=location, component_codes=component_codes,
+        headonly=headonly, verbose=verbose)
+
+
+class Reftek130(object):
+    def __init__(self):
+        self._data = np.array([], dtype=PACKET_FINAL_DTYPE)
+
+    def __str__(self, compact=True):
+        if compact:
+            info = [
+                "Reftek130 ({:d} packets)".format(len(self._data)),
+                "Packet Sequence  Byte Count  Data Fmt  Sampling Rate      "
+                "Time",
```

Comment:
The wrapping is a bit unfortunate here; I recommended making it a (private) class variable that's `format`ted in this method, so then it can be de-indented a bit.